src/ruby/spec/pb/codegen/package_option_spec.rb

```diff
@@ -0,0 +1,53 @@
+# Copyright 2018 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'spec_helper'
+require 'open3'
+require 'tmpdir'
+
+describe 'Code Generation Options' do
+  it 'should generate and respect package options' do
+    unless system('which grpc_ruby_plugin') && system('which protoc')
+      skip 'protoc || grpc_ruby_plugin missing, cannot verify package code-gen'
+    end
+
+    root_dir = File.join(File.dirname(__FILE__), '..', '..', '..', '..')
+    pb_dir = File.join(root_dir, 'proto')
+
+    # Regenerate it
+    plugin, = Open3.capture2('which', 'grpc_ruby_plugin')
+    plugin = plugin.strip
+    Dir.mktmpdir(nil, File.dirname(__FILE__)) do |tmp_dir|
+      pid = spawn(
+        'protoc',
+        '-I.',
+        'grpc/testing/package_options.proto',
+        "--grpc_out=#{tmp_dir}", # generate the service
+        "--ruby_out=#{tmp_dir}", # generate the definitions
+        "--plugin=protoc-gen-grpc=#{plugin}",
+        chdir: pb_dir)
+      Process.wait(pid)
+
+      begin
+        $LOAD_PATH.push(tmp_dir)
+
+        # this will fail if the package structure of the generated service does
+        # not match the generated protobuf defintions
+        expect(require('grpc/testing/package_options_services_pb')).to be_truthy
```

Comment:
nit: this check relies on some implicit things going on, e.g. that the protoc compiler generates the ruby message definitions with respect for the `ruby_package` option (correct?).

Can we increase the strictness of this check to make clear that the services and protobuf messages were both generated in the correct module? E.g., perhaps assert that the expected class exists only after loading the `_pb` file?